shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/ClangArgs.kt

```diff
@@ -37,146 +37,174 @@ class ClangArgs(private val configurables: Configurables) : Configurables by con
-                            "-Xclang", "-isystem$absoluteTargetSysRoot/include/libcxx",
-                            "-Xclang", "-isystem$absoluteTargetSysRoot/lib/libcxxabi/include",
-                            "-Xclang", "-isystem$absoluteTargetSysRoot/include/compat",
-                            "-Xclang", "-isystem$absoluteTargetSysRoot/include/libc")
-
-                is KonanTarget.ZEPHYR ->
-                    listOf("-target", targetArg!!,
-                        "-fno-rtti",
-                        "-fno-exceptions",
-                        "-fno-asynchronous-unwind-tables",
-                        "-fno-pie",
-                        "-fno-pic",
-                        "-fshort-enums",
-                        "-nostdinc",
-                        // TODO: make it a libGcc property? 
-                        // We need to get rid of wasm sysroot first.
-                        "-isystem $targetToolchain/../lib/gcc/arm-none-eabi/7.2.1/include",
-                        "-isystem $targetToolchain/../lib/gcc/arm-none-eabi/7.2.1/include-fixed",
-                        "-isystem$absoluteTargetSysRoot/include/libcxx",
-                        "-isystem$absoluteTargetSysRoot/include/libc"
-                        ) +
-                    (configurables as ZephyrConfigurables).constructClangArgs()
-            }
-            return result
-        }
-
-    val clangArgsSpecificForKonanSources
+    private val clangArgsSpecificForKonanSources
         get() = runtimeDefinitions.map { "-D$it" }
 
-    private val host = HostManager.host
-
-    private val binDir = when (host) {
+    private val binDir = when (HostManager.host) {
         KonanTarget.LINUX_X64 -> "$absoluteTargetToolchain/bin"
         KonanTarget.MINGW_X64 -> "$absoluteTargetToolchain/bin"
         KonanTarget.MACOS_X64 -> "$absoluteTargetToolchain/usr/bin"
         else -> throw TargetSupportException("Unexpected host platform")
     }
 
-    private val extraHostClangArgs =
-            if (configurables is GccConfigurables) {
-                listOf("--gcc-toolchain=${configurables.absoluteGccToolchain}")
-            } else {
-                emptyList()
+    // TODO: Use buildList
+    private val commonClangArgs: List<String> = mutableListOf<List<String>>().apply {
+        add(listOf("-B$binDir", "-fno-stack-protector"))
+        if (configurables is GccConfigurables) {
+            add(listOf("--gcc-toolchain=${configurables.absoluteGccToolchain}"))
+        }
+        if (configurables is TargetableConfigurables) {
+            add(listOf("-target", configurables.targetArg!!))
+        }
+    }.flatten()
+
+    private val osVersionMin: String
+            get() {
+                require(configurables is AppleConfigurables)
+                return configurables.osVersionMin
             }
 
-    val commonClangArgs = listOf("-B$binDir", "-fno-stack-protector") + extraHostClangArgs
+    val specificClangArgs: List<String> = when (target) {
```

Comment:
Probably should be private (e.g. to ensure nobody relies on this, since the contents are changed in this PR).